src/main/java/org/folio/helper/CheckinHelper.java

```diff
@@ -82,29 +83,49 @@ private Future<ReceivingResults> processCheckInPieces(CheckinCollection checkinC
   }
 
   private Future<Map<String, List<Piece>>> createItemsWithPieceUpdate(CheckinCollection checkinCollection, PiecesHolder holder, RequestContext requestContext) {
-    Map<String, List<CheckInPiece>> poLineIdVsCheckInPiece = getItemCreateNeededCheckinPieces(checkinCollection);
-    List<Future<Pair<String, Piece>>> futures = new ArrayList<>();
+    var poLineIdCheckInPieceMap = getItemCreateNeededCheckinPieces(checkinCollection);
+    var pieceFutures = new ArrayList<Future<PiecesHolder.PiecePoLineDto>>();
     return retrievePieceRecords(requestContext)
       .map(piecesFromStorage -> {
         holder.withPiecesFromStorage(piecesFromStorage);
-        piecesFromStorage.forEach((poLineId, pieces) -> poLineIdVsCheckInPiece.get(poLineId)
+        piecesFromStorage.forEach((poLineId, pieces) -> poLineIdCheckInPieceMap.get(poLineId)
           .forEach(checkInPiece -> pieces.forEach(piece -> {
+            var srcTenantId = piece.getReceivingTenantId();
+            var dstTenantId = checkInPiece.getReceivingTenantId();
             if (checkInPiece.getId().equals(piece.getId()) && Boolean.TRUE.equals(checkInPiece.getCreateItem())) {
-              futures.add(purchaseOrderLineService.getOrderLineById(poLineId, requestContext)
+              pieceFutures.add(purchaseOrderLineService.getOrderLineById(poLineId, requestContext)
                 .map(PoLineCommonUtil::convertToCompositePoLine)
-                .compose(compPOL -> pieceCreateFlowInventoryManager.processInventory(compPOL, piece,
-                  checkInPiece.getCreateItem(), requestContext))
-                .map(aVoid -> Pair.of(poLineId, piece)));
+                .compose(compPol -> pieceCreateFlowInventoryManager.processInventory(compPol, piece, checkInPiece.getCreateItem(), requestContext))
+                .map(voidResult -> new PiecesHolder.PiecePoLineDto(poLineId, piece)));
+            } else if (checkInPiece.getId().equals(piece.getId()) && Objects.nonNull(srcTenantId) && Objects.nonNull(dstTenantId) && !srcTenantId.equals(dstTenantId)) {
```

Comment:
InventoryUtils.allowItemRecreate can be uased instead of Objects.nonNull(srcTenantId) && Objects.nonNull(dstTenantId) && !srcTenantId.equals(dstTenantId)